### PR TITLE
Debian Package Testing and Build Fixes

### PR DESCRIPTION
**Pull request by brthor**
Changes to make the debian package build much more robust.

* Errors at any stage will fail the build.
* General Cleanup
* Don't do machine install of bats
* ~~Run E2E Tests against package installed bits~~ https://github.com/dotnet/cli/issues/381

/cc @Sridhar-MS @krwq @anurse @piotrpMSFT

### CODE REVIEW

File: scripts/_common.sh [hunk 1]
```diff
@@ -79,6 +87,7 @@ if [ -z "$RID" ]; then
     fi
 fi
 
+export REPOROOT=$(cd $COMMONDIR/.. && pwd)
```
Comment (anurse) [inline scripts/_common.sh]: I was originally going to do this but was getting weird paths. Having said that, I was also messing with other things at the same time. Glad to have this fixed up! :+1:

**Comment by anurse** [comment]
Looks good to me, :shipit: assuming the Windows failure isn't relevant.

File: scripts/_common.sh (hunk 1 above)
Comment (brthor) [inline scripts/_common.sh]: The DIR variable was overwriting the DIR in whatever script it got sourced in. Had me pretty confused for a bit too.

**Comment by brthor** [comment]
Windows is the Roslyn AV failure, going to get this merged.

**Pull request merged**